File: database_connection/task_list/task_list_java/src/main/java/com/repository/FileTaskRepository.java
package com.repository;

import java.util.List;
import java.util.Optional;

import com.model.Task;
import com.utils.FileHandler;

public class FileTaskRepository implements TaskRepository {

  private final FileHandler fileHandler;

  public FileTaskRepository(FileHandler fileHandler) {
    this.fileHandler = fileHandler;
  }

  @Override
  public void insert(Task task) {
    var tasks = findAll();
    task.setId(tasks.isEmpty() ? 1 : tasks.get(tasks.size() - 1).getId() + 1);
    tasks.add(task);
    fileHandler.writeObjectToFile(tasks);
  }

  @Override
  public void update(int id, Task task) {
    var tasks = findAll();
    tasks.set(id - 1, task);
    fileHandler.writeObjectToFile(tasks);
  }

  @Override
  public void deleteById(int id) {
    var tasks = findAll();
    tasks.remove(id - 1);
    fileHandler.writeObjectToFile(tasks);
  }

  @Override
  public Optional<Task> findById(int id) {
    var tasks = findAll();
    return Optional.ofNullable(tasks.get(id - 1));
  }

  @Override
  public List<Task> findAll() {
    return fileHandler.readObjectsFromFile();
  }
}

File: database_connection/task_list/task_list_python_venv/repository/file_task_repository.py
from model.task import Task
from repository.task_repository import TaskRepository
from utils.file_handler import FileHandler


class FileTaskRepository(TaskRepository):
  def __init__(self, file_handler: FileHandler) -> None:
    super().__init__()
    self.file_handler = file_handler

  def insert(self, task: Task):
    tasks = self.find_all()
    task.id = 1 if not tasks else tasks[len(tasks) - 1].id + 1
    tasks.append(task)
    self.file_handler.write_to_file(tasks)

  def update(self, id: int, task: Task):
    tasks = self.find_all()
    tasks[id - 1] = task
    self.file_handler.write_to_file(tasks)

  def delete_by_id(self, id: int):
    tasks = self.find_all()
    tasks.remove(tasks[id - 1])
    self.file_handler.write_to_file(tasks)

  def find_by_id(self, id: int) -> Task:
    tasks = self.find_all()
    return tasks[id - 1]

  def find_all(self) -> list[Task]:
    return self.file_handler.read_from_file()
